test/contracts/root-element.spec.tsx

```diff
@@ -0,0 +1,56 @@
+import React = require('react');
+import {findDOMNode} from 'react-dom';
+import {ClientRenderer, expect, RenderingContext} from 'test-drive-react';
+import * as WixReactComponents from '../../src';
+import {isReactComponent} from '../utils/is-react-component';
+
+// const allComponents = Object.keys(WixReactComponents);
+// Not all components are passing the contract test right now.
+// failing: TreeView, NumberInput, Toggle, BirthdayPicker, RadioButton, RadioGroup, Portal
+
+const testedComponents: string[] = ['DatePicker', 'CheckBox', 'Image', 'DropDown'];
+
+describe('Root Element contract', function() {
+    testedComponents.forEach(exportName => describe(exportName, function() {
+        const ComponentClass = (WixReactComponents as any)[exportName];
+        if (isReactComponent(ComponentClass)) {
+            assertRootElementContract(ComponentClass);
+        }
+    }));
+});
+
+export function assertRootElementContract(Component: React.ComponentType<any>): void {
+    const clientRenderer = new ClientRenderer();
+    function render<P>(element: React.ReactElement<P>): RenderingContext<P> & { rootNode: Element } {
+        const output = clientRenderer.render(element);
+        return {rootNode: findDOMNode(output.result as React.ReactInstance), ...output};
+    }
+
+    afterEach(() => {
+        clientRenderer.cleanup();
+    });
+
+    it('performs data-automation-id merge', function() {
+         const {select, rootNode} = render(<Component data-automation-id="CONTRACT_TEST"/>);
+         expect(select('CONTRACT_TEST'), 'data-automation-id not properly merged').to.equal(rootNode);
+    });
+
+    it('performs data-* attribute merge', function() {
+        const customValue = 'some-custom-value';
+        const {rootNode} = render(<Component data-some-custom-attr={customValue}/>);
+        expect(rootNode).to.have.attribute('data-some-custom-attr');
+        expect(rootNode.getAttribute('data-some-custom-attr')).to.contain(customValue);
+    });
+
+    it('performs inline style merge', function() {
+        const sampleColor = 'rgb(255, 0, 0)';
+        const {rootNode} = render(<Component style={{backgroundColor: sampleColor}}/>);
+        expect(getComputedStyle(rootNode).backgroundColor, 'inline style not properly merged').to.equal(sampleColor);
+    });
+
+    it('performs className merge', function() {
```

Comment:
Again (and ^^^) I can't fully test the functionality, only its "generic part" - I don't know what the component really does on its root element.